src/NuGet.Core/NuGet.Packaging/Signing/Signatures/Signature.cs

```diff
@@ -421,6 +423,21 @@ private static IReadOnlyList<Timestamp> GetTimestamps(SignerInfo signer, string
 
             return timestampList;
         }
+
+        private static void LogAdditionalContext(IX509Chain chain, List<SignatureLog> issues)
+        {
+            ILogMessage logMessage = chain.AdditionalContext;
+
+            if (logMessage is not null)
+            {
+                SignatureLog issue = SignatureLog.Issue(
+                    fatal: false,
+                    logMessage.Code,
+                    logMessage.Message);
+
+                issues.Add(issue);
+            }
+        }
```

Comment:
I see exact same logic in `src/NuGet.Core/NuGet.Packaging/Signing/Timestamp/Timestamp.cs` file change also. Can we move this function to some common place so that it can be invoked in both the places without having to duplicate the code?